blog/content/post/2020-04-26-measuring_memory_usage_rte.md

```diff
@@ -0,0 +1,217 @@
+`temp` file as an integer.  To get the output in a more accessible format
+of Celcius degrees we will use such command:
+```
+echo $((`cat /sys/class/thermal/thermal_zone0/temp | cut -c 1-2`)).$((`cat /sys/class/thermal/thermal_zone0/temp | cut -c 3-5`))
+```
+
+### Other tools that might be useful
+
+`armbianmonitor` is a simple CLI monitoring program for armbian system which
+shows CPU load, percentage usage of: cpu, sys (processes executing in kernel
+mode), usr (normal processes executing in user mode), nice (niced processes
+executing in user mode), io (waiting for I/O to complete), irq (servicing
+interrupts) and CPU temperature.
+
+`vmstat` reports information about processes, memory, paging, block IO, traps,
+disks and cpu activity. The first report produced gives averages since the last
+reboot. Additional reports give information on a sampling period of length
+delay. The process and memory reports are instantaneous in either case.
+## Case study - memory and cpu usage during running regression tests
+
+### Armbian
+
+`free` output before tests:
+
+```
+total  used  free  shared  buff/cache  available
+Mem:   242M   39M   41M     14M        162M       164M
+Swap:  127M    0B  127M
+```
+
+`free` output during tests:
+
+```
+total  used  free  shared  buff/cache  available
+Mem:   242M   65M   13M     14M        163M       137M
+Swap:  127M    0B  127M
+```
+
+```
+total  used  free  shared  buff/cache  available
+Mem:   242M   35M   43M     14M        163M       167M
+Swap:  127M    0B  127M
+```
+
+`top` output before tests:
+
+```
+load average: 0.00, 0.00, 0.00
+Tasks: 123 total,   1 running, 122 sleeping,   0 stopped,   0 zombie
+%Cpu(s):  0.3 us,  0.4 sy,  0.0 ni, 99.3 id,  0.0 wa,  0.0 hi,  0.0 si,  0.0 st
+KiB Mem :   248564 total,    41848 free,    40556 used,   166160 buff/cache
+KiB Swap:   131068 total,   131068 free,        0 used.   168392 avail Mem
+```
+
+`top` output during tests:
+
+```
+load average: 0.53, 0.30, 0.12
+Tasks: 132 total,   1 running, 131 sleeping,   0 stopped,   0 zombie
+%Cpu(s):  0.0 us,  0.0 sy,  0.0 ni, 99.9 id,  0.0 wa,  0.0 hi,  0.0 si,  0.0 st
+KiB Mem :   248564 total,    38644 free,    42900 used,   167020 buff/cache
+KiB Swap:   131068 total,   131068 free,        0 used.   165280 avail Mem
+```
+
+```
+load average: 0.03, 0.13, 0.10
+Tasks: 125 total,   1 running, 124 sleeping,   0 stopped,   0 zombie
+%Cpu(s):  0.2 us,  0.6 sy,  0.0 ni, 99.2 id,  0.0 wa,  0.0 hi,  0.0 si,  0.0 st
+KiB Mem :   248564 total,    44672 free,    36524 used,   167368 buff/cache
+KiB Swap:   131068 total,   131068 free,        0 used.   171404 avail Mem
+```
+
+#### Comparison table
+
+|                       |load   | CPU temp   | Mem usage    |
+|:----------------------|:-----:|:----------:|:------------:|
+|**before tests**       | 0.03  | 43.9&deg;C | 32MB / 242MB |
+|**during tests (max)** | 0.53  | 51&deg;C   | 65MB / 242MB |
+|**after tests**        | 0.12  | 44.2&deg;C | 33MB / 242MB |
```

Comment:
@pkonkol graphical diagram using Google Calc or OpenOffice would be appreciate and would say more to reader then plain numbers